debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/ui/CDebugUIPlugin.java

```diff
@@ -319,10 +328,29 @@ public void start(BundleContext context) throws Exception {
 			@Override
 			public IStatus runInUIThread(IProgressMonitor monitor) {
 				startupInUIThread();
+				// We need to enable Toggle Instruction Step Mode only if Debug UI plug-in is loaded.
+				setActivityEnabled(TISM_ACTIVITY_ID, true);
 				return Status.OK_STATUS;
 			}
 		};
 		wjob.schedule();
+
+		// Disable the activity when the platform shuts down.
+		IWorkbench workbench = PlatformUI.getWorkbench();
```

Comment:
Raghu, I meant **all**new code should go in the job.
The point is, the start() method of activator should do as less as possible for multiple reasons.